app/src/main/java/org/listenbrainz/android/ui/screens/login/LoginActivity.kt

```diff
@@ -1,112 +1,102 @@
 package org.listenbrainz.android.ui.screens.login
 
-import android.content.Intent
-import android.graphics.drawable.ColorDrawable
-import android.net.Uri
 import android.os.Bundle
-import android.view.Menu
 import android.widget.Toast
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
 import dagger.hilt.android.AndroidEntryPoint
-import org.listenbrainz.android.R
-import org.listenbrainz.android.databinding.ActivityLoginBinding
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
 import org.listenbrainz.android.model.AccessToken
 import org.listenbrainz.android.model.UserInfo
+import org.listenbrainz.android.repository.AppPreferencesImpl
 import org.listenbrainz.android.ui.components.ListenBrainzActivity
-import org.listenbrainz.android.ui.screens.dashboard.DashboardActivity
-import org.listenbrainz.android.util.LBSharedPreferences
+import org.listenbrainz.android.ui.theme.ListenBrainzTheme
+import org.listenbrainz.android.util.LBSharedPreferences.STATUS_LOGGED_OUT
 import org.listenbrainz.android.util.ListenBrainzServiceGenerator
 import org.listenbrainz.android.viewmodel.LoginViewModel
 
+/** ***NOTE:*** Always start this activity by passing a boolean extra with key **"startLogin"** as **true**.*/
 @AndroidEntryPoint
 class LoginActivity : ListenBrainzActivity() {
 
-    private var binding: ActivityLoginBinding? = null
-    private var loginViewModel: LoginViewModel? = null
+    private lateinit var viewModel: LoginViewModel
+    var isIntentLaunched : Boolean? = null
 
     public override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        binding = ActivityLoginBinding.inflate(layoutInflater)
-        setContentView(binding!!.root)
-        supportActionBar!!.setBackgroundDrawable(ColorDrawable(resources.getColor(R.color.app_bg)))
-        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
-        loginViewModel = ViewModelProvider(this)[LoginViewModel::class.java]
-        loginViewModel!!.accessTokenLiveData!!.observe(this) { accessToken: AccessToken? ->
-            saveOAuthToken(accessToken)
-        }
-        loginViewModel!!.userInfoLiveData!!.observe(this) { userInfo: UserInfo? ->
-            saveUserInfo(userInfo)
+    
+        // Controls launch of startLogin
+        isIntentLaunched = intent.getBooleanExtra("startLogin", false)
```

Comment:
Why keep this flag?

Reply:
This didn't do it. Whenever a user clicks _No thanks_ on the site, the `code` param is again `null`, which means redirecting back to the app will again launch auth and user will be stuck in this loop, unless they close the app manually.

Reply:
As I had mentioned earlier to you, we want the user to authenticate the app the first thing, hence after this PR is merged, I will look to move this auth part to the onboarding page. I don't want the user to use the app if they aren't authenticated. It doesn't make sense. Given that feed will be the first page the user interacts with, the user is supposed to be authenticated.

Reply:
Got rid of the flag :)